trimmer_decorator.rb

```diff
@@ -0,0 +1,6 @@
+require_relative 'decorator'
+class TrimmerDecorator < Decorator
+  def correct_name
+    nameable.correct_name[0, 10]
+  end
+end
```

Comment:
- [ ] To have control over this logic, it would be best to to use the length check to decide whether to trim the name which is a requirement based on this milestone.
- The above provides more control and flexibility as it allows you to handle the trimming logic based on the name's length. Your solution approach i assumes that `nameable` and `correct_name` are set up as described, which is not always the case.

### Requirement:
![image](https://github.com/yusufyusie/School-Library/assets/50990925/3338488f-f9f6-4f59-b7ba-a0b0facb38c5)

### Sample code:
```

class TrimmerDecorator < Decorator
    # The name has a maximum of 10 characters. If it's longer it should trim the word.
    def correct_name
      @nameable.correct_name.length > 10 ? @nameable.correct_name[0..9] : @nameable.correct_name
    end
end
```